CMake/External_OpenCV.cmake

```diff
@@ -1,59 +1,75 @@
+  if(fletch_ENABLE_FFmpeg)
+    message(STATUS "OpenCV depending on internal FFmpeg")
+    set(OpenCV_DEPENDS FFmpeg ${OpenCV_DEPENDS})
 
-  # OpenCV uses pkg-config to find libraries to link against and use, so placing
-  # our instal target library pkgconfig directory on the path link in order to
-  # take precedence.
-  if(NOT WIN32)
-    option(OpenCV_Enable_FFmpeg "" OFF)
-    if (OpenCV_Enable_FFmpeg)
-      # Setting ``cmake_command`` to add custom configuretion to CMAKE_ARGS generation
-      set(custom_cmake_command CMAKE_COMMAND PKG_CONFIG_PATH=${fletch_BUILD_INSTALL_PREFIX}/lib/pkgconfig:$ENV{PKG_CONFIG_PATH} ${CMAKE_COMMAND})
-      message(STATUS "Custom cmake comand for OpenCV: \"${custom_cmake_command}\"")
+    # OpenCV uses pkg-config to find libraries to link against and use, so placing
+    # our instal target library pkgconfig directory on the path link in order to
+    # take precedence.
+    if(NOT WIN32)
+      option(OpenCV_Enable_FFmpeg "" OFF)
+      if (OpenCV_Enable_FFmpeg)
+        # Setting ``cmake_command`` to add custom configuretion to CMAKE_ARGS generation
+        set(custom_cmake_command CMAKE_COMMAND PKG_CONFIG_PATH=${fletch_BUILD_INSTALL_PREFIX}/lib/pkgconfig:$ENV{PKG_CONFIG_PATH} ${CMAKE_COMMAND})
+        message(STATUS "Custom cmake comand for OpenCV: \"${custom_cmake_command}\"")
+      endif()
+    else()
+      message(WARNING "Custom linking of FFMPEG with OpenCV is undefined on Windows. OpenCV may correctly find the locally built FFmpeg, but it is not guaranteed.")
+      # TODO: Figure out how OpenCV finds ffmpeg on Windows.
     endif()
-  else()
-    message(WARNING "Custom linking of FFMPEG with OpenCV is undefined on Windows. OpenCV may correctly find the locally built FFmpeg, but it is not guaranteed.")
-    # TODO: Figure out how OpenCV finds ffmpeg on Windows.
   endif()
+endif(FALSE)
+
+# Handle GPU disable flag
+if(fletch_DISABLE_GPU_SUPPORT)
+  set(OpenCV_EXTRA_BUILD_FLAGS ${OpenCV_EXTRA_BUILD_FLAGS} -DWITH_CUBLAS=OFF -DWITH_CUDA=OFF -DWITH_CUFFT=OFF)
+  set(OpenCV_EXTRA_BUILD_FLAGS ${OpenCV_EXTRA_BUILD_FLAGS} -DWITH_OPENCL=OFF -DWITH_OPENCLAMDBLAS=OFF -DWITH_OPENCLAMDFFT=OFF)
+  set(OpenCV_EXTRA_BUILD_FLAGS ${OpenCV_EXTRA_BUILD_FLAGS} -DBUILD_opencv_gpu=OFF -DBUILD_opencv_ocl=OFF)
+endif()
+
+if (OpenCV_SELECT_VERSION VERSION_EQUAL 2.4.11)
+  set(OPENCV_PATCH_COMMAND ${CMAKE_COMMAND} -DOpenCV_patch:PATH=${fletch_SOURCE_DIR}/Patches/OpenCV -DOpenCV_source:PATH=${fletch_BUILD_PREFIX}/src/OpenCV -P ${fletch_SOURCE_DIR}/Patches/OpenCV/Patch.cmake)
+else()
+  set(OPENCV_PATCH_COMMAND "")
+endif()
+
+# Include link to contrib repo if enabled
+if (fletch_ENABLE_OpenCV_contrib)
+  set(OpenCV_CONTRIB_ARG "-DOPENCV_EXTRA_MODULES_PATH:PATH=${OpenCV_contrib_MODULE_PATH}")
+  list( APPEND OpenCV_DEPENDS OpenCV_contrib)
 endif()
 
 # Set Eigen dependency if we're locally building it
+option(fletch_ENABLE_EIGEN "Should Eigen Support be turned on for OpenCV?" ${_OpenCV_ENABLE_EIGEN_DEFAULT})
```

Comment:
That variable seems poorly named. I only moved it from the IF(FALSE) block. I don't see any reason to always pass Eigen to OpenCV if built. I will make that change.